### PR TITLE
Do not keep references to futures

### PR DESCRIPTION
**Pull request by mitar**

### CODE REVIEW

**Comment by kentonv** [comment]
Nice catch! But there could still be a leak if the future were to throw for some other reason than timing out (currently there's no code path that does this, but it's easy to imagine adding one). It seems like what we actually want is to wrap the `future.wait()` in a try/finally, where the finally block deletes the map entry and clears the timeout?

**Comment by mitar** [comment]
Updated.

File: server.js [hunk 1]
```diff
@@ -50,15 +50,18 @@ Meteor.methods({
 
     logins[token] = future;
 
-    var timeout = setTimeout(function () {
-      future.throw(new Meteor.Error("timeout", "Gave up waiting for login rendezvous XHR."));
-    }, 10000);
-
-    var info = future.wait();
+    try {
+      var timeout = setTimeout(function () {
```
Comment (kentonv) [inline server.js]: `timeout` is declared here and then used in the finally block, which violates lexical scoping, which technically works in Javascript but it's weird.

Can you have the `try` only wrap the `future.wait()` line? I don't think `setTimeout` will ever throw anyway.

Comment (mitar) [inline server.js]: Variables are global, so this is not really lexical scoping problem?

Comment (mitar) [inline server.js]: And, eh, I think I am spending more time on style changes here then anything beneficial. Feel free to turn things around to your style/preference.

**Comment by mitar** [comment]
I merged from master, because it was a conflict after you merged my other pull request.

I moved out `var` so that is more aligned with lexical scoping.

File: server.js (hunk 1 above)
Comment (kentonv) [inline server.js]: Relying on Javascript's lack of lexical scoping is really confusing and can lead to bugs if the code is later updated to ES6. Anyway, I committed 8052e99 which supersedes this.

**Comment by kentonv** [comment]
Superseded by 8052e99. Thanks for finding this!

**Pull request closed without merging**